scripts/tests/linux/tv_casting_test_sequences.py

```
@@ -0,0 +1,183 @@
+ATTRIBUTE_CURRENT_PLAYBACK_STATE = '0x0000_0000'  # Application Cluster Spec 6.10.6 Attribute ID: Current State of Playback
+
+
+class App(Enum):
+    """An enumeration of the supported applications."""
+
+    TV_APP = 'tv-app'
+    TV_CASTING_APP = 'tv-casting-app'
+
+
+class Step:
+    """A class to represent a step in a test sequence for validation.
+
+    A Step object contains attributes relevant to a test step where each object contains:
+    - `app` subprocess to parse for `output_msg` or `send input_cmd`
+    - `timeout_sec` specified the timeout duration for parsing the `output_msg` (optional, defaults to DEFAULT_TIMEOUT_SEC)
+    - `output_msg` or `input_cmd` (mutually exclusive)
+
+    Note: The first entry in the test sequence list of steps should always be the output string to verify that the tv-app 
+    is up and running.
+
+    For output message blocks, define the start line, relevant lines, and the last line. If the last line contains trivial closing 
+    characters (e.g., closing brackets, braces, or commas), include the line before it with actual content. For example:
+        `Step(subprocess_='tv-casting-app', output_msg=['InvokeResponseMessage =', 'exampleData', 'InteractionModelRevision =', '},'])`
+
+    For input commands, define the command string with placeholders for variables that need to be updated. For example:
+        `Step(subprocess_='tv-casting-app', input_cmd='cast request 0\n')`
+    """
+
+    def __init__(
+        self,
+        app: App,
+        timeout_sec: Optional[int] = DEFAULT_TIMEOUT_SEC,
+        output_msg: Optional[List[str]] = None,
+        input_cmd: Optional[str] = None,
+    ):
+        # Validate that either `output_msg` or `input_cmd` is provided, but not both.
+        if output_msg is not None and input_cmd is not None:
+            logging.error('Step cannot contain both `output_msg` and `input_cmd`. Either `output_msg` or `input_cmd` should be provided.')
+            sys.exit(1)
```

sys.exit is a bit too drastic in constructors. Can we raise an exception in these cases instead of doing a sys.exit here?

---

Addressed in the latest commit.